### PR TITLE
Fix issue when internal disk update would clean VM description

### PR DESCRIPTION
**Pull request by vbauzysvmware**
[Ref: #728](https://github.com/vmware/terraform-provider-vcd/issues/728)

This fix solves the issue when in terraform VM and internal disk resources are used and internal disk update happens (like resize). So on the second terrafom apply VM description is missing and an update on VM is initiated. Also which currently requires VM power off and power on.

### CODE REVIEW

**Comment by lvirbalas** [comment]
Looks like this PR contains other PRs, hence is not reviewable in this state.

**Comment by vbauzysvmware** [comment]
Ready for review

File: .changes/v2.14.0/418-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Fix Issue #728: `vm.UpdateInternalDisksAsync()` didn't send VM description and as result would delete it [GH-418]
```
Comment (lvirbalas) [inline .changes/v2.14.0/418-bug-fixes.md:1]:
```suggestion
* Fix Issue #728: `vm.UpdateInternalDisksAsync()` didn't send VM description and as a result would delete it [GH-418]
```

**Review by lvirbalas — COMMENTED** [review]
Interesting. Are there any other fields that we are not sending during disk update, which could have a similar effect?

File: .changes/v2.14.0/418-bug-fixes.md (hunk 1 above)
Comment (vbauzysvmware) [reply .changes/v2.14.0/418-bug-fixes.md:1]: updated

**Comment by vbauzysvmware** [comment]
> Interesting. Are there any other fields that we are not sending during disk update, which could have a similar effect?

I am not aware of any anymore.

**Review by lvirbalas — APPROVED** [review]

File: govcd/vm_test.go [hunk 2]
```diff
@@ -1087,6 +1091,11 @@ func (vcd *TestVCD) Test_UpdateInternalDisk(check *C) {
 	check.Assert(disk.BusNumber, Equals, diskSettings.BusNumber)
 	check.Assert(disk.AdapterType, Equals, diskSettings.AdapterType)
 
+	// verify VM description still available - bug fix test
```
Comment (lvirbalas) [inline govcd/vm_test.go:1094]:
```suggestion
	// verify VM description still available - test for bugfix #418
```

File: govcd/vm_test.go [hunk 3]
```diff
@@ -1052,6 +1052,10 @@ func (vcd *TestVCD) Test_UpdateInternalDisk(check *C) {
 	vm, storageProfile, diskSettings, diskId, previousProvisioningValue, err := vcd.createInternalDisk(check, vmName, 1)
 	check.Assert(err, IsNil)
 
+	description := "Test_UpdateInternalDisk_Description"
```
Comment (lvirbalas) [inline govcd/vm_test.go:1055]: Please also add a comment above of why this has to be done explicitly.

File: .changes/v2.14.0/418-bug-fixes.md (hunk 1 above)
Comment (dataclouder) [inline .changes/v2.14.0/418-bug-fixes.md:1]: It's not clear what "it" refers to. Also, from the conversation in Issue #728, it doesn't seem that this fix addresses that problem.

File: govcd/vm_test.go (hunk 2 above)
Comment (vbauzysvmware) [reply govcd/vm_test.go:1094]: ty. changed

File: govcd/vm_test.go (hunk 3 above)
Comment (vbauzysvmware) [reply govcd/vm_test.go:1055]: added

File: .changes/v2.14.0/418-bug-fixes.md (hunk 1 above)
Comment (vbauzysvmware) [reply .changes/v2.14.0/418-bug-fixes.md:1]: yes, it is not 100% as I had very limited information

Comment (vbauzysvmware) [reply .changes/v2.14.0/418-bug-fixes.md]: updated "it"

**Review by dataclouder — APPROVED** [review]

**Review by Didainius — APPROVED** [review]
Tests passed. Go ahead.

**Pull request merged**